### PR TITLE
Support echartsInstance.on(eventName: string, handler: Function)

### PR DESCRIPTION
**Pull request by chfw**
Enable event handler on all chart instances!

Here's the example code:

```
# coding=utf-8
from __future__ import unicode_literals

from pyecharts import Map
import pyecharts.events as events
from pyecharts_javascripthon.dom.functions import alert


def on_click(params):
    alert(params.name)

value = [155, 10, 66, 78]
attr = ["福建", "山东", "北京", "上海"]
map = Map("全国地图示例", width=1200, height=600)
map.add("", attr, value, maptype='china', is_label_show=True)
map.on(events.MOUSE_CLICK, on_click)
map.render()
```

Here is how it behaves:

### CODE REVIEW

File: docs/zh-cn/api.md [hunk 1]
```diff
@@ -99,6 +100,114 @@ js 文件仓库路径。可以设置本地或者远程地址。所有的远程
...
+**on(event_name, handler)**
+
+添加[事件处理函数](http://echarts.baidu.com/api.html#events)。
+
+请注意，事件处理函数是在浏览器里运行，但是要求你用 Python 写哦。
+
+这是支持的所有事件
+
+```
+# Mouse Events
```
Comment (kinegratii) [inline docs/zh-cn/api.md]: Please group these constants into different class for readability

File: pyecharts/constants.py [hunk 2]
```diff
@@ -25,3 +25,40 @@
     '531,42.034l175.125-42.917l175.125,42.917l0.531-42.034l-134'
     '.449-92.931l12.238-241.308L1705.06,1318.313z'
 }
+
+
+# Mouse Events
+
```
Comment (kinegratii) [inline pyecharts/constants.py]: Please group these constants into different class  or subpackage for readability

Comment (chenjiandongx) [reply pyecharts/constants.py]: 同意 @kinegratii  的观点

**Comment by kinegratii** [comment]
How about create new package `pyecharts.echarts` , and refactor with the following module moving.
```
pyecharts.option => pyecharts.echart.options
pyehcharts.events => pyecharts.echarts.events
pyecharts.style => pyecharts.echarts.style
```
Why?
1 `pyecharts.echarts` is the package for direct API of echarts ,The `option` and `events` are corresponding to the sections.
2 In architecture of pyecharts, `pyecharts.echarts` is low-api,which called by `pyecharts.charts` and `pyecharts.engine`. This is the beginning of rendering processor.

**Comment by kinegratii** [comment]
After this PR, the v0.5.0 will be `feature-frozen`,and is ready to release.

**Comment by chenjiandongx** [comment]
```
How about create new package pyecharts.echarts , and refactor with the following module moving.

pyecharts.option => pyecharts.echart.options
pyehcharts.events => pyecharts.echarts.events
pyecharts.style => pyecharts.echarts.style
Why?
1 pyecharts.echarts is the package for direct API of echarts ,The option and events are corresponding to the sections.
2 In architecture of pyecharts, pyecharts.echarts is low-api,which called by pyecharts.charts and pyecharts.engine. This is the beginning of rendering processor.
```
@kinegratii  感觉还可以归类出一个 `引擎` 包，把 engine.py, conf.py 转入里面

**Comment by chfw** [comment]
I would vote for pyecharts.echarts because option, events, style logically belongs to echarts, whereas engine.py is dealing with rendering, jinja2 module.

Let's make it in a separate PR.

**Pull request merged**